### PR TITLE
ReOrganized hipBLASLt samples

### PR DESCRIPTION
**Pull request by tsenwang**

### CODE REVIEW

File: clients/samples/ext_op/sample_hipblaslt_ext_op_amax.cpp
Comment (vin-huang) [inline clients/samples/ext_op/sample_hipblaslt_ext_op_amax.cpp:1]: please move this ext op sample to XX_amex_ext

File: clients/samples/ext_op/sample_hipblaslt_ext_op_amax_with_scale.cpp
Comment (vin-huang) [inline clients/samples/ext_op/sample_hipblaslt_ext_op_amax_with_scale.cpp:1]: please move this ext op sample to XX_amex_with_scale_ext

File: clients/samples/23_deprecated/sample_hipblaslt_gemm_ext_deprecated.cpp
Comment (vin-huang) [inline clients/samples/23_deprecated/sample_hipblaslt_gemm_ext_deprecated.cpp:1]: please remove the word "23_"

**Comment by tsenwang** [comment]
Screenshot 2024-08-27 132859

**Review by vin-huang — DISMISSED** [review]
LGTM

**Review by jichangjichang — APPROVED** [review]
lgtm

**Review by jichangjichang — CHANGES_REQUESTED** [review]
14_turing_wgm_gemm_ext ->14_tuning_wgm_gemm_ext

**Comment by tsenwang** [comment]
> 14_turing_wgm_gemm_ext ->14_tuning_wgm_gemm_ext

changed Thank You

**Comment by tsenwang** [comment]
root@banff-cyxtera-s73-1:/Workspace/hipBLASLt/build/release# make test
Running tests...
Test project /Workspace/hipBLASLt/build/release
      Start  1: sample_hipblaslt_gemm
 1/38 Test  #1: sample_hipblaslt_gemm ...............................   Passed    1.06 sec
      Start  2: sample_hipblaslt_gemm_ext
 2/38 Test  #2: sample_hipblaslt_gemm_ext ...........................   Passed    1.12 sec
      Start  3: sample_hipblaslt_gemm_batched
 3/38 Test  #3: sample_hipblaslt_gemm_batched .......................   Passed    1.84 sec
      Start  4: sample_hipblaslt_gemm_batched_ext
 4/38 Test  #4: sample_hipblaslt_gemm_batched_ext ...................   Passed    1.32 sec
      Start  5: sample_hipblaslt_gemm_tuning_splitk_ext
 5/38 Test  #5: sample_hipblaslt_gemm_tuning_splitk_ext .............   Passed    9.84 sec
      Start  6: sample_hipblaslt_gemm_bias
 6/38 Test  #6: sample_hipblaslt_gemm_bias ..........................   Passed    1.48 sec
      Start  7: sample_hipblaslt_gemm_bias_ext
 7/38 Test  #7: sample_hipblaslt_gemm_bias_ext ......................   Passed    1.25 sec
      Start  8: sample_hipblaslt_gemm_get_all_algos
 8/38 Test  #8: sample_hipblaslt_gemm_get_all_algos .................   Passed   10.05 sec
      Start  9: sample_hipblaslt_gemm_get_all_algos_ext
 9/38 Test  #9: sample_hipblaslt_gemm_get_all_algos_ext .............   Passed   10.84 sec
      Start 10: sample_hipblaslt_gemm_get_algo_by_index_ext
10/38 Test #10: sample_hipblaslt_gemm_get_algo_by_index_ext .........   Passed    9.97 sec
      Start 11: sample_hipblaslt_gemm_alphavec_ext
11/38 Test #11: sample_hipblaslt_gemm_alphavec_ext ..................   Passed    1.37 sec
      Start 12: sample_hipblaslt_gemm_gelu_aux_bias
12/38 Test #12: sample_hipblaslt_gemm_gelu_aux_bias .................   Passed    0.73 sec
      Start 13: sample_hipblaslt_gemm_gelu_aux_bias_ext
13/38 Test #13: sample_hipblaslt_gemm_gelu_aux_bias_ext .............   Passed    0.67 sec
      Start 14: sample_hipblaslt_gemm_amax
14/38 Test #14: sample_hipblaslt_gemm_amax ..........................   Passed    0.66 sec
      Start 15: sample_hipblaslt_gemm_amax_ext
15/38 Test #15: sample_hipblaslt_gemm_amax_ext ......................   Passed    1.03 sec
      Start 16: sample_hipblaslt_gemm_amax_with_scale
16/38 Test #16: sample_hipblaslt_gemm_amax_with_scale ...............   Passed    0.70 sec
      Start 17: sample_hipblaslt_gemm_amax_with_scale_ext
17/38 Test #17: sample_hipblaslt_gemm_amax_with_scale_ext ...........   Passed    0.75 sec
      Start 18: sample_hipblaslt_gemm_bgradb
18/38 Test #18: sample_hipblaslt_gemm_bgradb ........................   Passed    0.63 sec
      Start 19: sample_hipblaslt_gemm_ext_bgradb
19/38 Test #19: sample_hipblaslt_gemm_ext_bgradb ....................   Passed    0.66 sec
      Start 20: sample_hipblaslt_gemm_dgelu_bgrad
20/38 Test #20: sample_hipblaslt_gemm_dgelu_bgrad ...................   Passed    0.65 sec
      Start 21: sample_hipblaslt_gemm_dgelu_bgrad_ext
21/38 Test #21: sample_hipblaslt_gemm_dgelu_bgrad_ext ...............   Passed    0.67 sec
      Start 22: sample_hipblaslt_gemm_is_tuned_ext
22/38 Test #22: sample_hipblaslt_gemm_is_tuned_ext ..................   Passed    0.74 sec
      Start 23: sample_hipblaslt_gemm_tuning_wgm_ext
23/38 Test #23: sample_hipblaslt_gemm_tuning_wgm_ext ................   Passed    9.75 sec
      Start 24: sample_hipblaslt_gemm_with_scale_a_b
24/38 Test #24: sample_hipblaslt_gemm_with_scale_a_b ................   Passed   14.40 sec
      Start 25: sample_hipblaslt_gemm_with_scale_a_b_ext
25/38 Test #25: sample_hipblaslt_gemm_with_scale_a_b_ext ............   Passed    0.93 sec
      Start 26: sample_hipblaslt_groupedgemm_ext
26/38 Test #26: sample_hipblaslt_groupedgemm_ext ....................   Passed    2.49 sec
      Start 27: sample_hipblaslt_groupedgemm_fixed_mk_ext
27/38 Test #27: sample_hipblaslt_groupedgemm_fixed_mk_ext ...........   Passed   12.28 sec
      Start 28: sample_hipblaslt_groupedgemm_get_all_algos_ext
28/38 Test #28: sample_hipblaslt_groupedgemm_get_all_algos_ext ......   Passed   12.31 sec
      Start 29: sample_hipblaslt_gemm_mix_precision
29/38 Test #29: sample_hipblaslt_gemm_mix_precision .................   Passed    0.75 sec
      Start 30: sample_hipblaslt_gemm_mix_precision_ext
30/38 Test #30: sample_hipblaslt_gemm_mix_precision_ext .............   Passed    1.86 sec
      Start 31: sample_hipblaslt_gemm_mix_precision_with_amax_ext
31/38 Test #31: sample_hipblaslt_gemm_mix_precision_with_amax_ext ...   Passed    1.28 sec
      Start 32: sample_hipblaslt_gemm_attr_tciA_tciB
32/38 Test #32: sample_hipblaslt_gemm_attr_tciA_tciB ................   Passed    0.72 sec
      Start 33: sample_hipblaslt_ext_op_layernorm
33/38 Test #33: sample_hipblaslt_ext_op_layernorm ...................   Passed    0.57 sec
      Start 34: sample_hipblaslt_gemm_ext_deprecated
34/38 Test #34: sample_hipblaslt_gemm_ext_deprecated ................   Passed    1.18 sec
      Start 35: sample_hipblaslt_ext_op_amax
35/38 Test #35: sample_hipblaslt_ext_op_amax ........................   Passed    0.58 sec
      Start 36: sample_hipblaslt_ext_op_amax_with_scale
36/38 Test #36: sample_hipblaslt_ext_op_amax_with_scale .............   Passed    0.73 sec
      Start 37: hipblaslt-bench-groupedgemm-fixed-mk
37/38 Test #37: hipblaslt-bench-groupedgemm-fixed-mk ................   Passed  246.75 sec
      Start 38: hipblaslt-test
38/38 Test #38: hipblaslt-test ......................................   Passed  3093.69 sec

100% tests passed, 0 tests failed out of 38

Total Test time (real) = 3458.30 sec

**Review by vin-huang — APPROVED** [review]
LGTM

**Review by jichangjichang — APPROVED** [review]
lgtm

**Pull request merged**